src/fs/fs.c

```diff
@@ -370,59 +379,28 @@ int cfs_fs_create(const str256_t *fs_name)
 	bufstat.st_mtim.tv_sec = 0;
 	bufstat.st_ctim.tv_sec = 0;
 
-	RC_WRAP_LABEL(rc, free_info, kvtree_create, ns, (void *)&bufstat,
-	              sizeof(struct stat), &kvtree);
-
-	/* Attach kvtree pointer to cortxfs struct */
-	fs_node->cfs_fs.kvtree = kvtree;
+	RC_WRAP_LABEL(rc, delete_ns, kvtree_create, ns, (void *)&bufstat,
+		      sizeof(struct stat), &kvtree);
 
-	RC_WRAP_LABEL(rc, free_fs_node, kvtree_init, ns,
-		      fs_node->cfs_fs.kvtree);
-
-	root_node = malloc(sizeof(struct kvnode));
-	if (!root_node) {
-		rc = -ENOMEM;
-		log_err("Could not allocate memory for root kvnode");
-		goto free_fs_node;
-	}
-
-	*root_node = KVNODE_INIT_EMTPY;
-	RC_WRAP_LABEL(rc, free_fs_node, kvnode_load, kvtree, &(kvtree->root_node_id),
-		      root_node);
-	fs_node->cfs_fs.root_node = root_node;
-
-	RC_WRAP_LABEL(rc, free_fs_node, cfs_ino_num_gen_init,
+	RC_WRAP_LABEL(rc, delete_kvtree, fs_node_init, fs_node, ns, ns_size);
+	RC_WRAP_LABEL(rc, deinit_fs_node, cfs_ino_num_gen_init,
 		      &fs_node->cfs_fs);
 
-free_info:
-
-	if (rc == 0) {
-		LIST_INSERT_HEAD(&fs_list, fs_node, link);
-		goto out;
-	}
-
+	LIST_INSERT_HEAD(&fs_list, fs_node, link);
+	goto out;
+
+deinit_fs_node:
+	fs_node_deinit(fs_node);
+delete_kvtree:
+	kvtree_delete(kvtree);
+delete_ns:
+	ns_delete(ns);
 free_fs_node:
-	if (fs_node) {
-		if (fs_node->cfs_fs.ns)
-		{
-			free(fs_node->cfs_fs.ns);
-		}
-		if (root_node)
-		{
-			kvnode_fini(fs_node->cfs_fs.root_node);
-			free(fs_node->cfs_fs.root_node);
-		}
-		if (kvtree)
-		{
-			kvtree_fini(fs_node->cfs_fs.kvtree);
-			free(fs_node->cfs_fs.kvtree);
-		}
-
-		free(fs_node);
-	}
-
+	free(fs_node);
 out:
-	log_info("fs_name=" STR256_F " rc=%d", STR256_P(fs_name), rc);
+	if (rc != -ENOMEM) {
```

Comment:
Similar doubt about tracing condition and message content.

Reply:
same as previous comments.